### PR TITLE
Viewer advisory

### LINKED ISSUE #5670: Prototype/Wireframe development for "open with advisory"
# What is it and who's it for?
Experienced and new users browsing through the Wellcome collection who could come across content that is disturbing or explcit

# Implementation

* Modal window advisory that shows when opening the digitised content on a works page
* The advisory window will show before the viewer renders the digital content
* The modal window to close and show the digitised content when the user interacts with a "show contents" call to action
* Should the user choose not to show the content, then the back button will take them to the works page
* The contents of the modal window should advise that a cookie will be set and this will disable the advisory for the duration of their session 

## Testing and tracking
* Set up testing to monitor the usage and bouce rate from the CTA in the modal window
* Qualitative content research does language impact things
* Qual testing for content clarity

## Measurement
- Numbers of users that see the advisory as a blocker to using content

## Other considerations 
<---Delete as appropriate--->
- Impact on URL design
- Cardigan documentation

### PR DESCRIPTION
**Pull request by gestchild**
Fixes #5670 
access

### CODE REVIEW

File: catalogue/webapp/components/WorkDetails/WorkDetails.js [hunk 1]
```diff
@@ -262,28 +266,30 @@ const WorkDetails = ({
...
-              }}
-            >
-              <ButtonSolidLink
-                icon="eye"
-                text="View"
-                trackingEvent={{
-                  category: 'WorkDetails',
-                  action: 'follow view link',
-                  label: work.id,
+          {itemLinkState === 'useLibraryLink' &&
```
Comment (jamesgorrie) [inline catalogue/webapp/components/WorkDetails/WorkDetails.js:269]: What is this piece of logic trying to say?

File: catalogue/webapp/pages/item.js [hunk 2]
```diff
@@ -121,6 +155,38 @@ const ItemPage = ({
     ...sharedPaginatorProps,
   };
 
+  useEffect(() => {
+    setOrigin(`${window.location.protocol}//${window.location.hostname}`);
```
Comment (jamesgorrie) [inline catalogue/webapp/pages/item.js:159]: Are we missing the `:`

File: catalogue/webapp/components/WorkDetails/WorkDetails.js (hunk 1 above)
Comment (gestchild) [reply catalogue/webapp/components/WorkDetails/WorkDetails.js]: Sorry, this is horrible and shouldn't be here anyway. I've move the logic into the getItemLinkState function, which should be clearer

File: catalogue/webapp/test/components/__snapshots__/WorkDetails.test.js.snap [hunk 3]
```diff
@@ -135,7 +135,7 @@ exports[`Feature: 2. As a library member I want to request an item Scenario: I'm
           width={null}
         >
           <div
-            className="Space-sc-4az8w3-0 Modal__ModalWindow-sc-12234yr-2 qQBZz deVjSn shadow bg-white"
+            className="Space-sc-4az8w3-0 Modal__ModalWindow-sc-12234yr-2 qQBZz dHZhHz shadow bg-white font-black"
```
Comment (davidpmccormick) [inline catalogue/webapp/test/components/__snapshots__/WorkDetails.test.js.snap:138]: I vote to look at impoving/removing this snapshot test – it's nothing but trouble as it is at the moment.

File: common/utils/iiif.js [hunk 4]
```diff
@@ -24,6 +25,22 @@ export function getServiceId(currentCanvas: ?IIIFCanvas): ?string {
...
 
+export function getAuthService(iiifManifest: ?IIIFManifest): ?AuthService {
+  // $FlowFixMe
+  return iiifManifest?.service?.find(service => service.authService) || null;
+}
+
+export function getImageAuthService(canvas: ?IIIFCanvas) {
+  const serviceArray = canvas?.images?.[0]?.resource?.service?.[0]?.service;
+  const authService =
+    serviceArray &&
```
Comment (davidpmccormick) [inline common/utils/iiif.js:36]: `serviceArray?.find(...`

File: common/views/components/IIIFViewer/parts/MainViewer.js [hunk 5]
```diff
@@ -53,22 +66,41 @@ const ItemRenderer = memo(({ style, index, data }) => {
...
+    imageAuthService &&
+    imageAuthService.profile === 'http://iiif.io/api/auth/0/login/restricted';
   return (
     <div style={style}>
       {scrollVelocity === 3 || isProgrammaticScroll ? (
         <div style={{ display: 'flex', justifyContent: 'center' }}>
           <LL lighten={true} />
         </div>
+      ) : isRestricted ? (
+        <MessageContainer>
```
Comment (davidpmccormick) [inline common/views/components/IIIFViewer/parts/MainViewer.js:93]: should this `MessageContainer` be wrapped in `{imageAuthService && (…)}`, then the `h2` and `p` could also be conditionally rendered based on whether `label` and `description` are there?

File: common/views/components/Modal/Modal.js [hunk 6]
```diff
@@ -74,10 +75,10 @@ const ModalWindow = styled(Space).attrs({
   v: { size: 'xl', properties: ['padding-top', 'padding-bottom'] },
   h: { size: 'xl', properties: ['padding-left', 'padding-right'] },
   className: classNames({
-    'shadow bg-white': true,
+    'shadow bg-white font-black': true,
   }),
 })`
-  z-index: 1001;
+  z-index: 10001;
```
Comment (davidpmccormick) [inline common/views/components/Modal/Modal.js:81]: 💤

File: common/views/components/styled/Padlock.js [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+    width: 80%;
+    height: 62%;
+    left: 10%;
+    bottom: 0;
+    background-color: ${props => props.theme.color('yellow')};
+    border-radius: 15%;
+    z-index: 1;
+
+    /* keyhole */
+    ::before {
```
Comment (davidpmccormick) [inline common/views/components/styled/Padlock.js:57]: should this be `&::before` ? If not, what is it referring to?

**Review by davidpmccormick — COMMENTED** [review]
Looks good I think. Only a couple of questions/suggestions

**Review by davidpmccormick — APPROVED** [review]
Nice.

**Pull request merged**